File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Input_object.java
package com.github.dedis.popstellar.model.objects;

public class Input_object {
  private final String txOutHash;
  private final int txOutIndex;
  private final Script_input_object script;

  /**
   * @param txOutHash Previous (to-be-used) transaction hash
   * @param txOutIndex index of the previous to-be-used transaction
   * @param script The script describing the unlock mechanism
   */
  public Input_object(String txOutHash, int txOutIndex, Script_input_object script) {
    this.script = script;
    this.txOutHash = txOutHash;
    this.txOutIndex = txOutIndex;
  }

  public String get_tx_out_hash() {
    return txOutHash;
  }

  public int get_tx_out_index() {
    return txOutIndex;
  }

  public Script_input_object get_script() {
    return script;
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/InputObject.kt
package com.github.dedis.popstellar.model.objects

import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.objects.digitalcash.ScriptInputObject
import com.github.dedis.popstellar.model.objects.security.PublicKey

@Immutable
class InputObject
/**
 * @param txOutHash Previous (to-be-used) transaction hash
 * @param txOutIndex index of the previous to-be-used transaction
 * @param script The script describing the unlock mechanism
 */
(val txOutHash: String, val txOutIndex: Int?, val script: ScriptInputObject) {

  val pubKey: PublicKey
    get() = script.pubKey

  override fun toString(): String {
    return "InputObject{txOutHash='$txOutHash', txOutIndex=$txOutIndex, key=$pubKey}"
  }
}
